compiler/amd-comgr-stub/CMakeLists.txt

```diff
@@ -0,0 +1,108 @@
+# Copyright Advanced Micro Devices, Inc.
+# SPDX-License-Identifier:  MIT
+
+cmake_minimum_required(VERSION 3.25)
+
+project(amd-comgr-stub
+  LANGUAGES C CXX ASM
+  VERSION 1.0)
+
+include(CMakePackageConfigHelpers)
+
+set(AMD_COMGR_STAGE_PATH "" CACHE PATH "Path to the dist directory containing the amd-comgr project")
+set(IMPLIB_SCRIPT "${THEROCK_SOURCE_DIR}/build_tools/third_party/implib/implib-gen.py" CACHE PATH "Path to implib-gen.py")
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
+find_package(Threads REQUIRED)
+find_package(Python3 COMPONENTS Interpreter)
+
+if(WIN32)
+  message(FATAL_ERROR "amd-comgr-stub not supported/required on Windows")
+endif()
+
+message(STATUS "COMGR PATH: ${AMD_COMGR_STAGE_PATH}")
+if(NOT EXISTS "${AMD_COMGR_STAGE_PATH}")
+  message(FATAL_ERROR "Must set -DAMD_COMGR_STAGE_PATH to a valid install location of amd-comgr (got ${AMD_COMGR_STAGE_PATH})")
+endif()
+message(STATUS "IMPLIB-GEN: ${IMPLIB_SCRIPT}")
+if(NOT EXISTS "${IMPLIB_SCRIPT}")
+  message(FATAL_ERROR "Could not find implib-gen.py: ${IMPLIB_SCRIPT}")
+endif()
+
+# Generate stub files.
+set(TRAMPOLINE_DIR "${CMAKE_CURRENT_BINARY_DIR}/trampoline")
+execute_process(
+  COMMAND "${Python3_EXECUTABLE}" "${IMPLIB_SCRIPT}"
+    --outdir "${TRAMPOLINE_DIR}"
+    --dlopen-callback amd_comgr_stub_dlopen
+    --thread-safe
+    --lazy-load
+    "${AMD_COMGR_STAGE_PATH}/lib/libamd_comgr.so"
+  COMMAND_ERROR_IS_FATAL ANY
+)
+
+add_library(
+  amd_comgr
+  STATIC
+  "${TRAMPOLINE_DIR}/libamd_comgr.so.tramp.S"
+  "${TRAMPOLINE_DIR}/libamd_comgr.so.init.c"
+)
```

Comment:
I was going to suggest `.so.trampoline.S` for the file name (to avoid a possibly offensive abbreviation), but I see this is generated by `${IMPLIB_SCRIPT}`. If we do fork the code further we could make such changes.

Reply:
Never even thought about that. Not going to touch it.